scripts/naming_conventions_yaml_parser.py

```diff
@@ -0,0 +1,117 @@
+'''
+This is an experimental prototype that generates naming convention checkers from a .yaml file. Currently the templates
+used to identify the object being checked are stored in /afs/cern.ch/user/d/daho/yaml/templates.
+
+David Ho
+August 2016
+'''
+
+import yaml
+import argparse
+import os
+import datetime
+import re
+
+class ConventionReader(object):
+    """Reader class"""
+    def __init__(self, yamlfile):
+        super(ConventionReader, self).__init__()
+        self.yamlfile = yamlfile
+        self.rules = {}
+        self.project_name = ""
+        self.known_clang_types= {
+            "Function": "FunctionDecl",
+            "Type": "TypeDecl",
+            "DataMember": "FieldDecl",
+            "Const": "VarDecl",
+            "Enumerator": "EnumDecl",
+            "Namespace": "NamespaceDecl"
+        }
+        self.template_directory = "/afs/cern.ch/user/d/daho/yaml/templates/identifiers"
```

Comment:
something like this should do the trick: `self.template_directory, _ = os.path.split(os.path.realpath(__file__))`